### ISSUE TITLE
Unable to locate the specified class: Session.php

### ISSUE THREAD

**Issue by calvintam236** [bcit-ci/CodeIgniter#2826]
I just upgraded code from 2.1.4 to 3.0 develop branch, after all, there is problem when i autoload the session driver on specific page.

This error only occur on one page, only for http auth. Other pages work fine (I checked the sessions generated).

I checked 'session' is in the right location(drivers) in the autoload file.

Controller code:

``` php
class Auth extends CI_Controller
{
    function index()
    {
        $this->load->model('users');
        if (!!$this->input->server('AUTH_TYPE') && $this->users->check($this->input->server('PHP_AUTH_USER'), sha1($this->input->server('PHP_AUTH_PW'))))
        {
            $this->session->set_userdata('version', 'internal');
            redirect(site_url());
        }
        else
        {
            $this->output->set_header('WWW-Authenticate: Basic realm="authorized personnel only"');
            set_status_header(401);
        }
    }
}
```

**Comment by anoopkp06** [comment]
Hello narfbg ,

I have followed the upgrade instructions correctly and installed successfully,  Amd my session also works fine, But when i try for a 404 page, it give me same issue 

Unable to locate the specified class: Session.php 

Thanks 
Anoop

**Comment by narfbg** [comment]
What do you mean by "when i try for a 404 page"?

**Comment by anoopkp06** [comment]
404 page means,  a page or controller not exists

**Comment by ZombieProtectionAgency** [comment]
I think Narf knows what a 404 is. I think what he meant was how are you trying for a 404? Are you just going to a page that doesn't exist or are your trying to force a 404 in the code somehow?

**Comment by anoopkp06** [comment]
Hi

I m trying some text on my url to make a 404 and i already set 404 page in routes.php 

When the 404 page loads, i m getting this error 

Unable to locate the specified class: Session.php

thanks 
Anoop

**Comment by narfbg** [comment]
Can't reproduce.

**Comment by simplifiedotnet** [comment]
Just trying to help out. So I decided to dive v3 today and got the same issue when I called the constructor inside my library. One thing I missed in the docs is not to use CI_Controller as the base class of your custom library and to use get_instance() as replacement for $this:

//controllers/ExampleController.php
class ExampleController extends CI_Controller{
	public function __construct(){parent::__construct();}
	public final function show(){$this->load->library('samplelibrary');}
}

//libraries/SampleLibrary.php
class SampleLibrary extends CI_Controller{
	public function __construct(){
	//Extending CI_Controller as base class triggers the "Unable to locate the specified class: Session.php" error message.
		parent::__construct();  
	}
}

Obviously I made a mistake on the structure there, but you might want to have proper handling of these weird scenarios.
